src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1222,8 +1226,12 @@ void TorrentHandle::toggleSequentialDownload()
 
 void TorrentHandle::setFirstLastPiecePriority(bool b)
 {
-    if (!hasMetadata()) return;
+    if (!hasMetadata()) {
+        m_needSetFirstLastPiecePriority = b;
```

Comment:
@glassez I was trying to avoid using that flag as the canonical indicator as to whether a torrent has first/last piece priority set, since then we need to worry about it going out of sync with the actual state of the torrent. (Note that the current implementation of `hasFirstLastPiecePriority()` doesn't use such a flag but instead checks if the first and last pieces actually have the correct priority set, so there's nothing to get out of sync.)

However, if we need to store it in the resume data I do believe we'll need to use a flag that way.

Also, we need to be careful about setting the flag unconditionally, because we don't want the torrent to have first/last priority if it doesn't contain any media files that support it. So I'll maintain that behavior in how `m_hasFirstLastPiecePriority` is set.